File: week0/fctrl.cpp
#include <iostream>

using namespace std;

int main() {
    int num_cases;
    cin >> num_cases;

    int n;
    while (num_cases-- > 0) {
        cin >> n;
        int ans = 0;

        for (int d = 5; d <= n; d *= 5) {
            ans += n / d;
        }

        cout << ans << '\n';
    }

    return 0;
}

File: week0/FCTRL.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int numCases = scan.nextInt();
        for (int i = 0; i < numCases; i++) {
            int n = scan.nextInt();

            int ans = 0;
            for (int j = 5; j <= n; j *= 5) {
                ans += n / j;
            }

            System.out.println(ans);
        }
    }
}
